pkg/actions/repo/enable.go

```diff
@@ -354,6 +374,96 @@ func mergeMaps(m1 map[string][]byte, m2 map[string][]byte) map[string][]byte {
 	return result
 }
 
+// PublicKey represents a public SSH key as it is returned by flux
+type PublicKey struct {
+	Key string `json:"key"`
+}
+
+func getPublicKeyFromFlux(ctx context.Context, namespace string, timeout time.Duration, restConfig *rest.Config,
+	cs kubeclient.Interface) (PublicKey, error) {
+	var deployKey PublicKey
+	try := func(rootURL string) error {
+		fluxURL := rootURL + "api/flux/v6/identity.pub"
+		req, reqErr := http.NewRequest("GET", fluxURL, nil)
+		if reqErr != nil {
+			return fmt.Errorf("failed to create request: %s", reqErr)
+		}
+		repoCtx, repoCtxCancel := context.WithTimeout(ctx, timeout)
+		defer repoCtxCancel()
+		req = req.WithContext(repoCtx)
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
+			return fmt.Errorf("failed to query Flux API: %s", err)
+		}
+		if resp.Body == nil {
+			return fmt.Errorf("failed to fetch Flux deploy key from: %s", fluxURL)
+		}
+		defer resp.Body.Close()
+
+		jsonErr := json.NewDecoder(resp.Body).Decode(&deployKey)
+		if jsonErr != nil {
+			return fmt.Errorf("failed to decode Flux API response: %s", jsonErr)
+		}
+
+		if deployKey.Key == "" {
+			return fmt.Errorf("failed to fetch Flux deploy key from: %s", fluxURL)
+		}
+		return nil
+	}
+	err := portForward(namespace, "flux", 3030, "Flux", restConfig, cs, try)
+	return deployKey, err
+}
+
+type tryFunc func(rootURL string) error
+
+func portForward(namespace string, nameLabelValue string, port int, name string,
+	restConfig *rest.Config, cs kubeclient.Interface, try tryFunc) error {
+	fluxSelector := metav1.LabelSelector{
```

Comment:
very nitpicky but `gofmt` isn't opinionated enough here IMO, I usually do long argument list formatting like:

```
functionName(
    argument, argument, argument, argument, argument, argument,
    argument, argument, argument, argument, argument, argument, 
) error {
```
to separate the header from body but it's not very important either way

Reply:
yep can change it. this was one of the things i moved from elsewhere